### PR TITLE
Update release notes draft to version v1.23.0-beta.0

### PR DESCRIPTION
**Pull request by Damans227**
**What type of PR is this?**
/kind documentation

**What this PR does / why we need it**:
This PR updates the Release Notes Draft to k/k v1.23.0-beta.0

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:
This is an automated PR generated from `krel The Kubernetes Release Toolbox`

### CODE REVIEW

File: releases/release-1.23/release-notes/maps/pr-103095-map.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+pr: 103095
+releasenote:
+  text: Add support for `PodAndContainerStatsFromCRI` featuregate, which allows a
```
Comment (AuraSinis) [inline releases/release-1.23/release-notes/maps/pr-103095-map.yaml:3]: Should it be feature gate?

**Comment by cici37** [comment]
The deletion from release-notes-draft.md sounds a little wired to me. Would you mind to keep your si-release master up to date and try it again?

File: releases/release-1.23/release-notes/maps/pr-103095-map.yaml (hunk 1 above)
Comment (reylejano) [reply releases/release-1.23/release-notes/maps/pr-103095-map.yaml:3]: Yes, please see the discussion around this:
https://groups.google.com/g/kubernetes-sig-architecture/c/Nxsc7pfe5rw/m/vF2djJh0BAAJ
This discussion is in a file modified by this PR
https://github.com/kubernetes/kubernetes/pull/103095/files

**Comment by Damans227** [comment]
> The deletion from release-notes-draft.md sounds a little wired to me. Would you mind to keep your si-release master up to date and try it again?

Thanks for catching that @cici37. I am gonna close this one out and re run krel after updating my repos.

**Comment by Damans227** [comment]
Closed due to an error.

**Pull request closed without merging**